tests/ui/pin-ergonomics/pin_v2-attr.rs

```diff
@@ -0,0 +1,144 @@
+    // FIXME: delegation macros are not tested yet (how to?)
+    #[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on delegations
+    reuse <Struct as std::any::Any>::type_id;
+
+    #[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on inherent methods
+    fn method() {}
+}
+
+#[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on trait impl blocks
+impl Trait for Enum {
+    #[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on trait methods in impl blocks
+    fn method() {}
+}
+
+#[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on extern crates
+extern crate alloc;
+
+#[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on use statements
+use std::pin::Pin;
+
+#[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on statics
+static STATIC: () = ();
+
+#[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on constants
+const CONST: () = ();
+
+#[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on functions
+fn f<T, U>(#[pin_v2] param: Foo<T, U>)
+//~^ ERROR `#[pin_v2]` attribute cannot be used on function params
+//~| ERROR allow, cfg, cfg_attr, deny, expect, forbid, and warn are the only allowed built-in attributes in function parameters
+where
+    #[pin_v2]
+    //~^ ERROR `#[pin_v2]` attribute cannot be used on where predicates
+    T:,
+{
+    #[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on closures
+    || ();
+    #[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on expressions
+    [(), (), ()];
+    #[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on statements
+    let _: Foo<(), ()> = Foo::StructVariant {
+        #[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on struct fields
+        field: (),
+    };
+    match param {
+        #[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on match arms
+        Foo::UnitVariant => {}
+        Foo::TupleVariant(..) => {}
+        Foo::StructVariant {
+            #[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on pattern fields
+            field,
+        } => {}
+    }
+}
+
+#[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on modules
+mod m {}
+
+#[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on foreign modules
+extern "C" {
+    #[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on foreign types
+    type ForeignTy;
+
+    #[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on foreign statics
+    static EXTERN_STATIC: ();
+
+    #[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on foreign functions
+    fn extern_fn();
+}
+
+#[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on type alias
+type Type = ();
+
+#[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on macro defs
+macro_rules! macro_def {
+    () => {};
+}
+
+#[pin_v2] //~ ERROR `#[pin_v2]` attribute cannot be used on macro calls
+macro_def!();
+
+std::arch::global_asm! {
+    "{}",
+    #[pin_v2] //~ ERROR this attribute is not supported on assembly
+    const 0
+}
```

Comment:
This has to be limited to architectures on which it's supported.

(Thanks to @ehuss for spotting this.)

Reply:
I don't (quickly) find how to restrict the UI test to specific target archs only, and `#[cfg(target_arch = "xxx")]` would make different errors on different archs that would fail the test, so I removed it as it is not quite important here.